src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -240,4 +249,49 @@ private void removeCGFromMonitors(CertificateGenerator gen) {
         cisShadowMonitor.removeFromMonitor(gen);
     }
 
+
+    private RetryUtils.RetryConfig getRetryConfig() {
+        List<Class> retriableExceptions = Arrays.asList(
+                ThrottlingException.class,
+                InternalServerException.class,
+                AccessDeniedException.class);
+
+        return RetryUtils.RetryConfig.builder()
+                .initialRetryInterval(Duration.ofSeconds(3))
+                .maxAttempt(Integer.MAX_VALUE)
+                .retryableExceptions(retriableExceptions)
+                .build();
+    }
+
+    /**
+     * Uploads the stored certificates to the cloud.
+     *
+     * @param thingName  Core device name
+     *
+     * @throws CertificateEncodingException  If unable to get certificate encoding
+     * @throws KeyStoreException if unable to retrieve the certificate
+     * @throws IOException If unable to read certificate
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public void uploadCoreDeviceCAs(String thingName) throws CertificateEncodingException, KeyStoreException,
+            IOException {
+        List<String> certificatePemList = getCACertificates();
+
+        PutCertificateAuthoritiesRequest request =
+                PutCertificateAuthoritiesRequest.builder().coreDeviceThingName(thingName)
+                        .coreDeviceCertificates(certificatePemList).build();
+        try {
+            RetryUtils.runWithRetry(getRetryConfig(),
+                    () -> clientFactory.getGreengrassV2DataClient().putCertificateAuthorities(request),
+                    "put-core-ca-certificate", logger);
+        } catch (InterruptedException e) {
+            logger.atError().cause(e).log("Put core CA certificates got interrupted");
```

Comment:
when it is an interruption, don't log the cause. There's only 1 cause for an interruptedexception, so logging `cause(e)` just adds noise.

Also, this probably shouldn't even be an error anyway. Shutting down the upload would be relatively normal thing to happen, not an error.

Reply:
Agreed. This is the code we have in prod but I think this is a great opportunity to address it.